### PR TITLE
Display quickicons only for active components

### PR DESCRIPTION
**Pull request by chmst**
Pull Request for Issue #40684 .
When components are disabled, they don't show up in the components menu, but in quick icos. 
This PR hides also quickicons for disabled components.

### Summary of Changes
Check if a component is enabled before rendering a quicicon.

### Testing Instructions
see #40684

Activate quickicons for unprotected components ( tags, banners, contacts, newsfeeds, finder, workflows, associations)
Check if they are displayed. 
Then disable these components.
Check again - the quickicons ar no longer displayed.
If you re-activates the components, they come back.

### Actual result BEFORE applying this Pull Request
Quickicons to disabled components are displayed

### Expected result AFTER applying this Pull Request
Quickicons to disabled components are not displayed

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by brianteeman** [comment]
https://github.com/joomla/joomla-cms/issues/40684#issuecomment-1569822945

**Comment by Kostelano** [comment]
This does not work for workflow. Double-check empirically - turn on the integration, then the icon will disappear. Also, the component is protected and cannot be disabled in extension management. You need a condition - to enable / disable integration in the settings of the article component, and not enable / disable the component in principle.

File: administrator/modules/mod_quickicon/src/Helper/QuickIconHelper.php [hunk 1]
```diff
@@ -366,7 +367,7 @@ public function getButtons(Registry $params, CMSApplication $application = null)
             if ($params->get('show_languages')) {
                 $tmp = [
                     'image'   => 'icon-comments langmanager',
-                    'link'    => Route::_('index.php?option=com_languages&view=languages'),
```
Comment (Kostelano) [inline administrator/modules/mod_quickicon/src/Helper/QuickIconHelper.php]: Update the thread please. Correct `index.php?option=com_languages&view=languages`

Comment (chmst) [reply administrator/modules/mod_quickicon/src/Helper/QuickIconHelper.php]: Thanks for the hint.

**Comment by chmst** [comment]
@brianteeman I think it is better to make quicicons in the same way as the main menu.
If the menu doesn't show disabled components, why whould we show them in quickicons?

**Comment by chmst** [comment]
@Kostelano workflow indeed is a special case. We have one icon for workflow - but no specification for which component the worfkow is enabled.

**Comment by Kostelano** [comment]
> workflow indeed is a special case. We have one icon for workflow - but no specification for which component the worfkow is enabled.

So what does your new condition do for the workflow? The component is protected and cannot be disabled. The condition should be removed in my opinion.

By analogy with menu items - we do not need to show something that is not enabled (in this case, not integrated in the article settings). The same applies to the multilingual component, but this component can still be disabled, so the condition for it is correct.

**Comment by Kostelano** [comment]
I have tested this item :white_check_mark: successfully on 7a71c7e50b23961298b816fed7f88f878b765c0e<br /><br />After the changes I send a successful test, it works. And it makes sense.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40715">issues.joomla.org/tracker/joomla-cms/40715</a>.</sub>

**Comment by brianteeman** [comment]
> @brianteeman I think it is better to make quickicons in the same way as the main menu. If the menu doesn't show disabled components, why should we show them in quickicons?

By that logic the option to set them should also be removed from the module

**Comment by chmst** [comment]
We don't remove the menu links from the main menu, if a component is disabled, we just don't show the link if the componet is disabled. 
Removing the options from the module needs more programming, but you could make an alternative PR. Will be happy to test.

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 7a71c7e50b23961298b816fed7f88f878b765c0e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40715">issues.joomla.org/tracker/joomla-cms/40715</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40715">issues.joomla.org/tracker/joomla-cms/40715</a>.</sub>

**Comment by obuisard** [comment]
Thank you Christiane @chmst for the PR!

**Pull request merged**